core/src/main/java/org/jdbi/v3/tweak/StreamCallback.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.tweak;
+
+import java.util.stream.Stream;
+
+/**
+ * Callback for use with {@link org.jdbi.v3.ResultBearing#withStream(StreamCallback)}
+ */
+@FunctionalInterface
+public interface StreamCallback<T, R> {
```

Comment:
I'm all for moving them out.
I think the intention of the `tweak` package were for things that modified the behavior of statement preparation / execution (plugin-like things).
This certainly doesn't fit under that definition.

Maybe we can get rid of or give a better name to `tweak` entirely for v3.